Zend/zend_compile.c

```diff
@@ -9447,10 +9455,28 @@ static void zend_compile_const_decl(zend_ast *ast) /* {{{ */
 		name_node.op_type = IS_CONST;
 		ZVAL_STR(&name_node.u.constant, name);
 
-		zend_emit_op(NULL, ZEND_DECLARE_CONST, &name_node, &value_node);
+		last_op = zend_emit_op(NULL, ZEND_DECLARE_CONST, &name_node, &value_node);
 
 		zend_register_seen_symbol(name, ZEND_SYMBOL_CONST);
 	}
+	if (attributes_ast == NULL) {
+		return;
+	}
+	// Validate: attributes can only be applied to one constant at a time
+	// Since we store the AST for the attributes in the list of children,
+	// there should be exactly 2 children
+	if (list->children > 2) {
+		zend_error_noreturn(
+			E_COMPILE_ERROR,
+			"Cannot apply attributes to multiple constants at once"
+		);
+	}
```

Comment:
While I agree that this is a useful restriction to prevent misleading code, it is inconsistent with class constants: https://3v4l.org/XEvad

```
<?php

class Foo {
    #[\Deprecated]
    public const Foo = 1, Bar = 2;
}

var_dump(Foo::Foo, Foo::Bar);
```

Reply:
This was primarily a function of not being quite sure how to pass the attributes up to the runtime if applied to multiple constants at once - given that this is a new feature, I think its better to help prevent misleading code. What about adding this as a secondary question to the RFC?

Reply:
@TimWolla I just want to bump this - the current patch and RFC both restrict adding attributes to only one constant at a time, if you are opposed to that restriction please bring it up on the internal thread and we can see what other people think, it should be simple to make it a secondary RFC question if there are others who would prefer this restriction be removed